File: 0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root==null)return true;
        return checkLR(root.left,root.right);
    }
    public boolean checkLR(TreeNode l, TreeNode r){
        if(l ==null && r== null)return true;
        if(l==null || r == null)return false;
        if(l.val!=r.val)return false;
        return checkLR(l.left,r.right) && checkLR(l.right, r.left);
    }
}

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        
        q = deque()
        if root.left:
            q.append(root.left)
        if root.right:
            q.append(root.right)
        if len(q) == 1:
            return False
        while q:
            no_nodes = len(q)
            current_level_nodes = []
            for nodes in range(no_nodes):
                current_node = q.popleft()
                current_level_nodes.append(current_node.val if current_node else None)
                if current_node:
                    q.append(current_node.left if current_node.left else None)
                    q.append(current_node.right if current_node.right else None)
            
            if current_level_nodes != current_level_nodes[::-1]:
                return False

        return True
        